### PR TITLE
Resolve clusterIP issues from Helm3's 3-way match

### LINKED ISSUE #187: spec.clusterIP: Invalid value: \"None\"
Hi,

Running helm chart version: 14.3.0

```
    web:
      service:
        workerGateway:
          type: LoadBalancer
          annotations:
```
Error I'm seen
```
info="error updating the resource \"concourse-web\":\n\t cannot patch \"concourse-web\" with kind Service: Service \"concourse-web\" is invalid: spec.clusterIP: Invalid value: \"None\": field is immutable" targetNamespace=concourse release=concourse
```

Regards

### PR DESCRIPTION
**Pull request by JohnLBevan**
# Existing Issue
Fixes #187.

# Contributor Checklist
- [✓ ] Which branch are you merging into?
    - `master` is for changes related to the current release of the `concourse/concourse:latest` image and should be good to publish immediately

# Reviewer Checklist
> This section is intended for the core maintainers only, to track review progress. Please do not
> fill out this section.
- [ ] Code reviewed
- [ ] Topgun tests run
- [ ] Back-port if needed
- [ ] Is the correct branch targeted? (`master` or `dev`)

### CODE REVIEW

File: templates/web-api-svc.yaml [hunk 1]
```diff
@@ -26,11 +26,11 @@ spec:
     - {{ . }}
     {{- end }}
   {{- end }}
-  {{- if and (eq "ClusterIP" .Values.web.service.api.type) .Values.web.service.api.clusterIP }}
-  clusterIP: {{ .Values.web.service.api.clusterIP }}
+  {{- if and (eq "ClusterIP" .Values.web.service.api.type) (ne "None" .Values.web.service.api.clusterIP) }}
```
Comment (aoldershaw) [inline templates/web-api-svc.yaml:29]: I'm not sure this is quite what we want. `None` is actually a special value that indicates the service should be headless. I'm not sure why we made it the default, but it's clearly causing problems, and I don't think we need these services to be headless.

I think an approach similar to #163 is preferable, which changes the default to `nil`. Let me know if you disagree

Comment (JohnLBevan) [reply templates/web-api-svc.yaml:29]: Thank-you @aoldershaw.  Yes, that makes sense; apologies; I'm new to this whole tech stack and had assumed that None was just another null / nil / nothing alternative.  I've now read up and understand that:
- `None` = "Don't use 1 IP for this group of pods, give me all the members' IPs" 
- `nil` = "I don't care what the specific IP for this group is; assign it any free one"

**Comment by Jasstkn** [comment]
@JohnLBevan hi! Why did you close this PR?

**Comment by JohnLBevan** [comment]
@Jasstkn sorry for the slow response; only just discovered the GitHub inbox!

> @JohnLBevan hi! Why did you close this PR?

My proposed fix wasn't appropriate (see review comments from @aoldershaw on 8th Dec).

**Pull request closed without merging**